### PR TITLE
Add Windows support

### PR DESCRIPTION
**Pull request by traversaro**
Based on https://github.com/conda-forge/casadi-feedstock/pull/45 by @pariterre, done as new PR to quickly iterate on CI results.

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by traversaro** [comment]
@conda-forge-admin, please rerender

**Comment by traversaro** [comment]
@conda-forge-admin, please rerender

File: recipe/bld.bat [hunk 1]
```diff
@@ -1,22 +1,23 @@
...
+    -DCMAKE_PREFIX=lib/cmake/casadi^
+    -DLIB_PREFIX=lib^
+    -DBIN_PREFIX=bin^
     -DWITH_SELFCONTAINED=OFF^
     -DWITH_PYTHON=ON^
     -DWITH_PYTHON3=ON^
     -DWITH_LAPACK=ON^
     -DWITH_IPOPT=ON^
-    -DWITH_THREAD=ON^
+    -DWITH_THREAD=OFF^
```
Comment (pariterre) [inline recipe/bld.bat:20]: The python prefix should point to site package, and not root, you can have a look at the build.sh

Comment (traversaro) [reply recipe/bld.bat:20]: Fixed in https://github.com/conda-forge/casadi-feedstock/pull/47/commits/48575e87853e3f09ad5315637d8a5e470a3f40ac .

**Comment by traversaro** [comment]
The error: 
~~~
FAILED: casadi_nlpsol_ipopt.dll casadi_nlpsol_ipopt.lib 
cmd.exe /C "cd . && %BUILD_PREFIX%\Library\bin\cmake.exe -E vs_link_dll --intdir=casadi\interfaces\ipopt\CMakeFiles\casadi_nlpsol_ipopt.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100177~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100177~1.0\x64\mt.exe --manifests  -- C:\PROGRA~2\MICROS~1\2017\ENTERP~1\VC\Tools\MSVC\1416~1.270\bin\Hostx64\x64\link.exe /nologo casadi\interfaces\ipopt\CMakeFiles\casadi_nlpsol_ipopt.dir\ipopt_interface.cpp.obj casadi\interfaces\ipopt\CMakeFiles\casadi_nlpsol_ipopt.dir\ipopt_nlp.cpp.obj casadi\interfaces\ipopt\CMakeFiles\casadi_nlpsol_ipopt.dir\ipopt_interface_meta.cpp.obj  /out:casadi_nlpsol_ipopt.dll /implib:casadi_nlpsol_ipopt.lib /pdb:casadi_nlpsol_ipopt.pdb /dll /version:0.0 /machine:x64 /INCREMENTAL:NO -LIBPATH:%PREFIX%\Library\lib casadi.lib  ipopt.lib  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib  && cd ."
LINK: command "C:\PROGRA~2\MICROS~1\2017\ENTERP~1\VC\Tools\MSVC\1416~1.270\bin\Hostx64\x64\link.exe /nologo casadi\interfaces\ipopt\CMakeFiles\casadi_nlpsol_ipopt.dir\ipopt_interface.cpp.obj casadi\interfaces\ipopt\CMakeFiles\casadi_nlpsol_ipopt.dir\ipopt_nlp.cpp.obj casadi\interfaces\ipopt\CMakeFiles\casadi_nlpsol_ipopt.dir\ipopt_interface_meta.cpp.obj /out:casadi_nlpsol_ipopt.dll /implib:casadi_nlpsol_ipopt.lib /pdb:casadi_nlpsol_ipopt.pdb /dll /version:0.0 /machine:x64 /INCREMENTAL:NO -LIBPATH:%PREFIX%\Library\lib casadi.lib ipopt.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFEST /MANIFESTFILE:casadi_nlpsol_ipopt.dll.manifest" failed (exit code 1181) with the following output:
LINK : fatal error LNK1181: cannot open input file 'ipopt.lib'
~~~
was due to another error in ipopt's .pc file, that will be fixed in https://github.com/conda-forge/ipopt-feedstock/pull/61 .

**Comment by traversaro** [comment]
The compilation of the C++ library now works fine, there is now a linking error in the Python bindings:
~~~
2021-02-06T21:50:46.4767586Z FAILED: _casadi.pyd 
2021-02-06T21:50:46.6017555Z cmd.exe /C "cd . && %BUILD_PREFIX%\Library\bin\cmake.exe -E vs_link_dll --intdir=swig\python\CMakeFiles\_casadi.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100177~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100177~1.0\x64\mt.exe --manifests  -- C:\PROGRA~2\MICROS~1\2017\ENTERP~1\VC\Tools\MSVC\1416~1.270\bin\Hostx64\x64\link.exe /nologo swig\python\CMakeFiles\_casadi.dir\__\casadiPYTHON_wrap.cxx.obj  /out:_casadi.pyd /implib:_casadi.lib /pdb:_casadi.pdb /dll /version:0.0 /machine:x64 /INCREMENTAL:NO -LIBPATH:%SRC_DIR%\swig\..\build\lib %PREFIX%\libs\python36.lib  casadi.lib  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib  && cd ."
2021-02-06T21:50:46.7703375Z LINK: command "C:\PROGRA~2\MICROS~1\2017\ENTERP~1\VC\Tools\MSVC\1416~1.270\bin\Hostx64\x64\link.exe /nologo swig\python\CMakeFiles\_casadi.dir\__\casadiPYTHON_wrap.cxx.obj /out:_casadi.pyd /implib:_casadi.lib /pdb:_casadi.pdb /dll /version:0.0 /machine:x64 /INCREMENTAL:NO -LIBPATH:%SRC_DIR%\swig\..\build\lib %PREFIX%\libs\python36.lib casadi.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFEST /MANIFESTFILE:_casadi.pyd.manifest" failed (exit code 1120) with the following output:
2021-02-06T21:50:46.8512713Z    Creating library _casadi.lib and object _casadi.exp
2021-02-06T21:50:46.8516706Z casadiPYTHON_wrap.cxx.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: static class casadi::Matrix<double> __cdecl casadi::GenericExpression<class casadi::Matrix<double> >::_copysign(class casadi::Matrix<double> const &,class casadi::Matrix<double> const &)" (__imp_?_copysign@?$GenericExpression@V?$Matrix@N@casadi@@@casadi@@SA?AV?$Matrix@N@2@AEBV32@0@Z) referenced in function "class casadi::Matrix<double> __cdecl casadi::_copysign(class casadi::Matrix<double> const &,class casadi::Matrix<double> const &)" (?_copysign@casadi@@YA?AV?$Matrix@N@1@AEBV21@0@Z)
~~~

**Comment by traversaro** [comment]
A quick search revealed that probably there is some macro name collision related to `copysign`/`_copysign`, and a workaround can be activated with an appropriate CMake option: https://github.com/casadi/casadi/blob/3.5.5/CMakeLists.txt#L310 .

**Comment by pariterre** [comment]
Amazing job so far!!

I don't know if it is fix now, by you may will need to add the last line of the build.sh as well 
```bash
# Change a line of code that won't work with new version of python
sed -i'.original' 's/_object/object/' $SP_DIR/casadi/casadi.py
```
They inherite from _object which is no longer a thing in Python, It is now only object. So I parse the python.py file and simply replace the _object to object. But it was like 6 months ago, so maybe it is now fixed and no longer required!

**Comment by traversaro** [comment]
> I don't know if it is fix now, by you may will need to add the last line of the build.sh as well

Thanks! At the moment the problem seems that library are installed in the wrong location, probably because I set them to relative path in CMake (even if https://github.com/robotology/robotology-superbuild/blob/d0ff1c8ec9cb639486043c1cc559ad049622c5d0/cmake/Buildcasadi.cmake#L19 we do the same and there is working fine).

**Comment by traversaro** [comment]
Changing from:
~~~
     -DINCLUDE_PREFIX=include^ 
~~~
to
~~~
     -DINCLUDE_PREFIX:PATH=include^ 
~~~
did the trick, not sure why (I copied it from https://github.com/robotology/robotology-superbuild/blob/d0ff1c8ec9cb639486043c1cc559ad049622c5d0/cmake/Buildcasadi.cmake#L19, originally done by @GiulioRomualdi ).

**Comment by traversaro** [comment]
The PR is ready for review.

**Comment by pariterre** [comment]
What a great work! That is well appreciated !
I plan to add back the Threading in another PR. So I don't loose too much time, what was the reason you turned it off in the first place?

**Review by pariterre — APPROVED** [review]
LGTM :)

**Pull request merged**